### DISCUSSION TITLE
Question on how to setup.

### DISCUSSION THREAD

**Discussion by ebo-bpi** [ovh/cds#6300] [Q&A] [answered]
We have  a monorepo (nx / next) from which we build several applications. 
reading the documentation concepts we could build this in multiple ways:
1. create a pipeline with a job for every app.(checkout, build)
2. create  workflow with a pipeline and reuse this to build all apps. 

What is in your opinion the way we should set this up?
We build from 1 source, so I would like to reuse the checkout if possible. (reduce waiting time and io)
End result is a docker container which should be deployed in a later stage. (other pipeline or workflow?)

I would like to do a build for all affected apps (nx feature) and add this as a result status to the pr.
With multiple pipelines this will result in multiple pr comments for every app. (did not find a way to do this at workflow level) 

Hope you guys could give me some hints what the best approach would be

**Comment by yesnault** [comment, marked as answer]
Hi,

We suggest you to use only one workflow. We build CDS like this (which contains multiples apps as cdsctl, engine, ui, etc...)

The workflow:
<img width="424" alt="Capture d’écran 2022-09-22 à 14 58 58" src="https://user-images.githubusercontent.com/395454/191753400-06ab9008-d5d8-4e83-b2ea-1ad7d20af4ea.png">

The `build-all-cds pipeline`:
<img width="977" alt="Capture d’écran 2022-09-22 à 14 59 32" src="https://user-images.githubusercontent.com/395454/191753510-46a2f2da-7b9f-46ea-aaa1-7959f3288828.png">
This pipeline build the binaries in the first stage, then make the docker image in the second, then run integration tests...

If you need to customize the PR comment, you can `worker export` a variable and use it in the VCS Notification PR Body.

**Reply by ebo-bpi** [reply]
nice and clear answer. I'm gonna try this setup!
